From 9ba3a4ba686b3a619abfbdc1372aaa59a6328ee2 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 8 Nov 2015 10:08:56 -0500
Subject: [PATCH 0710/1936] Remove Typed Message Code

This interface, while nicer than directly using the C api (IMO), uses
template metaprogramming which older compilers cannot handle.
As this is a largely cosmetic difference, this code will be removed for
now, though it may be reintroduced once compilers have matured.
---
 src/Misc/MiddleWare.cpp | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index ad59ce63..b3265dac 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -8,7 +8,6 @@
 #include <rtosc/undo-history.h>
 #include <rtosc/thread-link.h>
 #include <rtosc/ports.h>
-#include <rtosc/typed-message.h>
 #include <lo/lo.h>
 
 #include <unistd.h>
@@ -387,7 +386,6 @@ struct ParamStore
 namespace Nio
 {
     using std::get;
-    using rtosc::rtMsg;
     rtosc::Ports ports = {
         {"sink-list:", 0, 0, [](const char *, rtosc::RtData &d) {
                 auto list = Nio::getSinks();
@@ -404,13 +402,13 @@ namespace Nio
         {"source::s", 0, 0, [](const char *msg, rtosc::RtData &d) {
                 if(rtosc_narguments(msg) == 0)
                     d.reply(d.loc, "s", Nio::getSource().c_str());
-                else if(rtMsg<const char*> m{msg})
-                    Nio::setSource(get<0>(m));}},
+                else
+                    Nio::setSource(rtosc_argument(msg,0).s);}},
         {"sink::s", 0, 0, [](const char *msg, rtosc::RtData &d) {
                 if(rtosc_narguments(msg) == 0)
                     d.reply(d.loc, "s", Nio::getSink().c_str());
-                else if(rtMsg<const char*> m{msg})
-                    Nio::setSink(get<0>(m));}},
+                else
+                    Nio::setSink(rtosc_argument(msg,0).s);}},
     };
 }
 
-- 
2.47.0

